application/src/test/java/com/sanction/lightning/resources/FacebookResourceTest.java

```diff
@@ -120,27 +126,92 @@ public void testGetMediaBytesWithNullUrl() {
   }
 
   @Test
-  public void testGetMediaByesWithBadUrl() {
-    when(urlDownloadService.fetchUrlConnection(any(String.class))).thenReturn(null);
+  public void testGetMediaByesWithNullUrlConnection() {
+    when(urlService.fetchUrlConnection(any(String.class))).thenReturn(null);
     Response response = resource.getMediaBytes(key, "Test");
 
     assertEquals(response.getStatusInfo(), Response.Status.BAD_REQUEST);
   }
 
   @Test
-  public void testGetMediaBytesWithBadConnection() {
-    when(urlDownloadService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
-    when(urlDownloadService.inputStreamToByteArray(any(URLConnection.class))).thenReturn(null);
+  public void testGetMediaBytesWithNullInputStream() {
+    when(urlService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
+    when(urlService.fetchInputStreamFromConnection(urlConnection)).thenReturn(null);
+
+    Response response = resource.getMediaBytes(key, "Test");
+
+    assertEquals(response.getStatusInfo(), Response.Status.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void testGetMediaBytesWithNullByteArray() {
+    when(urlService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
+    when(urlService.fetchInputStreamFromConnection(urlConnection)).thenReturn(inputStream);
+    when(urlService.inputStreamToByteArray(any(InputStream.class))).thenReturn(null);
```

Comment:
Do you want to make this `when(urlService.inputStreamToByteArray(inputStream)).thenReturn(null);` since you on the line above you made that method return `inputStream`?